app/UI/app_update.tsx

```diff
@@ -277,10 +280,79 @@ const handle_update_event = async (chan: PutChannel<true>, args: {
                     chan.put(true);
                 }
             });
-        } else if (event.type == "UpdateMessageFiles") {
-            console.log("to be implemented");
-        } else if (event.type == "UpdateEditorText") {
-            console.log("to be implemented");
+        } else if (event.type == "UploadImages") {
+            function validateImageFile(file: File) {
+                // https://nostr.build/.well-known/nostr/nip96.json
+                const max_byte_size = 10485760;
+                const image_postfix = [".png", ".jpg", ".jpeg", ".gif", ".webp"];
+                if (file.size > max_byte_size) {
+                    return false;
+                }
+                if (!image_postfix.some((postfix) => file.name.endsWith(postfix))) {
+                    return false;
+                }
+                return true;
+            }
+            async function uploadValidFiles(app: any, validFiles: File[], apiURL: string): Promise<{
+                uploadedURLs: string[];
+                errorMessages: string[];
+            }> {
+                const uploadedURLs: string[] = [];
+                const errorMessages: string[] = [];
+                const uploadPromises = validFiles.map(async (file) => {
+                    try {
```

Comment:
which part of this function throws?

Reply:
Adjusted the code, this part of the suggestion is outdated. Thanks anyway.